scripts/src2dest.js

```diff
@@ -0,0 +1,209 @@
+  if (match !== null) {
+    dictData.metadata.referencedWorks = [match[1]]
+  } else {
+    console.error('Cannot parse a referenced works field')
+  }
+
+  match = metadataStr.match(/# CC-CEDICT can be downloaded from:\n# (.+)/)
+  if (match !== null) {
+    dictData.metadata.downloadURI = match[1]
+  } else {
+    console.error('Cannot parse a download URI field')
+  }
+
+  match = metadataStr.match(/# Additions and corrections can be sent through:\n# (.+)/)
+  if (match !== null) {
+    dictData.metadata.editorURI = match[1]
+  } else {
+    console.error('Cannot parse an editor URI field')
+  }
+
+  match = metadataStr.match(/# For more information about CC-CEDICT see:\n# (.+)/)
+  if (match !== null) {
+    dictData.metadata.referenceURI = match[1]
+  } else {
+    console.error('Cannot parse a reference URI field')
+  }
+
+  match = metadataStr.match(/#! version=(.+)/)
+  if (match !== null) {
+    dictData.metadata.originalVersion = match[1]
+  } else {
+    console.error('Cannot parse an original version number field')
+  }
+
+  match = metadataStr.match(/#! subversion=(.+)/)
+  if (match !== null) {
+    dictData.metadata.originalSubversion = match[1]
+  } else {
+    console.error('Cannot parse an original subversion number field')
+  }
+
+  match = metadataStr.match(/#! format=(.+)/)
+  if (match !== null) {
+    dictData.metadata.originalFormat = match[1]
+  } else {
+    console.error('Cannot parse an original format field')
+  }
+
+  match = metadataStr.match(/#! charset=(.+)/)
+  if (match !== null) {
+    dictData.metadata.originalCharset = match[1]
+  } else {
+    console.error('Cannot parse an original charset field')
+  }
+
+  match = metadataStr.match(/#! publisher=(.+)/)
+  if (match !== null) {
+    dictData.metadata.publisher = match[1]
+  } else {
+    console.error('Cannot parse a publisher field')
+  }
+
+  match = metadataStr.match(/#! date=(\d{4})-(\d{2})-(\d{2})/)
+  if (match !== null) {
+    targetInfo.version = `v${match[1]}${match[2]}${match[3]}`
+    dictData.metadata.version = targetInfo.version
+  } else {
+    console.error('Cannot parse a version field')
+  }
+
+  match = metadataStr.match(/#! time=(.+)/)
+  if (match !== null) {
+    dictData.metadata.dateTime = new Date(parseInt(match[1], 10) * 1000)
+  } else {
+    console.error('Cannot parse a time field')
+  }
+
+  const metadataSize = encoder.encode(JSON.stringify(dictData.metadata, ...stringifyOptions)).length
+  let chunks = [] // eslint-disable-line prefer-const
+  let size = metadataSize
+  let currentChunkIndex = 0
+  sizeData.forEach((entry) => {
+    if ((size + entry.size) >= chunkSizeLimit) {
+      // This is a split point
+      if (currentChunkIndex === 0) {
+        // This is the first chunk and it must include a metadata object
+        chunks.push({
+          metaData: dictData.metadata,
+          entries: dictData.entries.slice(currentChunkIndex, entry.index - 1)
+        })
+      } else {
+        chunks.push({
```

Comment:
I think it would make sense to put the metadata in each file

Reply:
Great idea! Will do